### ISSUE TITLE
<iostream>: Wrong reading of float values

### ISSUE THREAD

**Issue by Fedr** [microsoft/STL#3980]
# Describe the bug
"1e-07" is read from std::istream as `1e-06`

# Command-line test case
```
#include <iostream>
#include <sstream>
using namespace std;

int main()
{
    istringstream istr("1e-07");
    double xx = 0;
    istr >> xx;
    cout << "xx = " << xx << endl;
    return 0;
}
```

# STL version
Visual Studio 2022 17.7
Please use Runtime Library not from DLL

# Additional context
First found here: https://stackoverflow.com/q/76937234/7325599
This is also DevCom-10450662 / VSO-1876474 / AB#1876474

**Comment by achabense** [comment]
https://github.com/microsoft/STL/blob/60f18856c56389001c38a9929ef37bd5c01c4e47/stl/inc/xlocnum#L1022-L1024
Here, it's eagerly adding an '"0"' before adding 'e'.

**Comment by Fedr** [comment]
Thanks. Could you please comment why DLL-runtimes of the same Visual Studio do not contain the bug? Is it because they were built from an older version of Microsoft STL?

**Comment by achabense** [comment]
> Thanks. Could you please comment why DLL-runtimes of the same Visual Studio do not contain the bug? Is it because they were built from an older version of Microsoft STL?

I guess so. There was an overhaul to the function in #3364, the branch may became redundant after that pr.

**Comment by fsb4000** [comment]
>Is it because they were built from an older version of Microsoft STL?

@Fedr Yes. Look at: https://github.com/microsoft/STL/wiki/Changelog#vc-redist-lockdown

>To reduce the frequency of required reboots when updating VS (and VS Previews), we "lock down" the Visual C++ Redistributable containing the STL's msvcp140.dll. (Most of the STL is header-only, so most changes are unaffected.) Approximately twice a year, specific VS Previews will be "unlocked", so the VC Redist will contain updated DLLs. VS 2022 17.8 Preview 2 will be the next unlocked release.

**Comment by fsb4000** [comment]
@CaseyCarter , I found the same DevCom bug: https://developercommunity.visualstudio.com/t/v143-1437-istringstream-incorrectly-con/10450662 , so it should also be closed after merge the fix.

**Comment by CaseyCarter** [comment]
Thanks @fsb4000, I've linked everything up.

This is a fairly nasty regression, we should consider backporting the fix to 17.7 / 17.8.
